include/exadg/grid/grid_motion_elasticity.h

```diff
@@ -94,7 +94,8 @@ class GridMotionElasticity : public GridMotionBase<dim, Number>
       pde_operator->initialize_dof_vector(rhs);
       pde_operator->compute_rhs_linear(rhs, time);
 
-      auto const iter = pde_operator->solve_linear(displacement, rhs, 0.0 /* no mass term */, time);
+      auto const iter = pde_operator->solve_linear(
+        displacement, rhs, 0.0 /* no mass term */, time, param.update_preconditioner);
```

Comment:
I am not sure if this is what we want to have (of course, it would then also be a bug in the nonlinear case above): I think we want to trigger the update of the preconditioner for the ALE mesh motion solver only every `N_update_interval`th time step (using a logic similar as in the single-field time integration schemes). This logic is not implemented inside `pde_operator->solve_(non)linear()`, right? Does this imply that we update the preconditioner for every ALE mesh motion solve (in every FSI iteration) if `param.update_preconditioner == true`? This could become very expensive.

Reply:
the grid mapping is constructed from the initial configuration (Lagrangian frame) right?
There are several aspects to this:
1) is the "mesh stiffness" updated depending on the actual displacement or are the parameters computed once initially and kept constant?
2) time step size might vary
3) nonlinear mesh motion model might be used

`pde_opereator->solve_(non)linear()` updates if the passed flag is `true` and in the nonlinear case additionally at Newton convergence or every _n_th Newton step.
As the mapping is a critical ingredient in practical applications, I would actually leave the choice to the user and allow setting the latter two flags in the nonlinear case + a time step logic like you mentioned. So I would  recommend using the same logic as in `time_int_gen_alpha.cpp`, line 149:

`  bool const update_preconditioner =
    this->param.update_preconditioner &&
    ((this->time_step_number - 1) % this->param.update_preconditioner_every_time_steps == 0);`

`pde_operator->solve_nonlinear()` takes care of the rest :100: 

Agree?

Reply:
Thinking about this again, I realized that the current grid motion techniques are (i) all solving stationary problems, (ii) the Lagrangian reference frame is considered as base and (iii) the parameters for "grid stiffening" do not depend on the solution itself, so for poisson and linear elasticity operaetors, we should deactivate the preconditioner update completely, shouldn't we? For the nonlinear elasticity case it makes sense though ...
Can you confirm (i)-(iii) @nfehn ?